File: apps/mybookkeeper/frontend/src/shared/types/welcome-manual/welcome-manual-section-field-response.ts
/**
 * Mirrors backend `WelcomeManualSectionFieldResponse`. A field is a simple
 * label/value pair (e.g. "Wi-Fi network" → "Lakeview") ordered within a
 * section. ``value`` is nullable — a label with no value yet is allowed.
 */
export interface WelcomeManualSectionFieldResponse {
  id: string;
  section_id: string;
  label: string;
  value: string | null;
  display_order: number;
  created_at: string;
}

File: apps/mybookkeeper/backend/app/schemas/welcome_manuals/welcome_manual_section_field_response.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WelcomeManualSectionFieldResponse(BaseModel):
    id: uuid.UUID
    section_id: uuid.UUID
    label: str
    value: str | None = None
    display_order: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
